kernel/src/process.rs

```diff
@@ -559,7 +560,7 @@ pub trait Process {
     ///   process's memory region.
     /// - [`Error::KernelError`] if there was an internal kernel error. This is
     ///   a bug.
-    fn sbrk(&self, increment: isize) -> Result<*const u8, Error>;
+    fn sbrk(&self, increment: isize) -> Result<CapabilityPtr, Error>;
```

Comment:
Same comment as on `brk` above.

Reply:
This one is simpler. The comment here is a bit sparse, but perhaps taking the libc documentation for sbrk (which this is meant to adhere to) makes it simpler to understand:

> On success, sbrk() returns the previous program break. (If the break was increased, then this value is a pointer to the start of the newly allocated memory).

That is, the value returned from `sbrk` should be a `CapabilityPtr` pointing to the previous heap break, with validity until the the end of the new heap (which should be at least length `increment` from the previous `sbrk`). It's fine (and convenient in practice) for it to be a region that is also valid starting from the beginning of the heap.

Reply:
Yes, sbrk needs to return the previous break to be compatible with what libc expects (and this was the previous behavior).

On the CHERI front CHERI does _not_ support merging of capabilities, and so the implementation in process standard is:

```
let base = self.mem_start() as usize;
let break_result = CapabilityPtr::new_with_metadata(
    old_break as *const (),
    base,
    (new_break as usize) - base,
    MetaPermissions::ReadWrite,
);
```

Which is a capability for the region `mem_start` to the _new_ break, with an address of the old break.

 I will amend the comment to say what the returned pointer should have authority for.

Reply:
This is great, thanks!